application/src/main/java/org/thingsboard/server/service/query/DefaultEntityQueryService.java

```diff
@@ -251,14 +251,13 @@ public DeferredResult<ResponseEntity> getKeysByQuery(SecurityUser securityUser,
         }
 
         if (isAttributes) {
-            Map<EntityType, List<EntityId>> typesMap = ids.stream().collect(Collectors.groupingBy(EntityId::getEntityType));
-            List<ListenableFuture<List<String>>> futures = new ArrayList<>(typesMap.size());
-            typesMap.forEach((type, entityIds) -> futures.add(dbCallbackExecutor.submit(() -> attributesService.findAllKeysByEntityIds(tenantId, type, entityIds))));
-            attributesKeysFuture = Futures.transform(Futures.allAsList(futures), lists -> {
-                if (CollectionUtils.isEmpty(lists)) {
+            ListenableFuture<List<String>> future;
+            future = dbCallbackExecutor.submit(() -> attributesService.findAllKeysByEntityIds(tenantId, ids));
```

Comment:
Ids need to be grouped by entity type.
Also, join declaration and assignment :)